lib/Charts/d3Sync.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+const childNodesSelector = function() {
+    return this.childNodes;
+};
+
+/**
+ * Function that synchronises children of a DOM element with javascript array of elements.
+ * If you are using D3, you are doing this all the time.
+ * Let's make that code more readable by having a common function we call for this purpose.
+ */
+function d3Sync(parent, arrayData, childElementTagName, updateCallback, transition=undefined)
+{
+    // remove stray elements
+    parent
+        .selectAll(childNodesSelector)
+        .filter(function() { return this.tagName !== childElementTagName; })
+        .remove();
```

Comment:
I just tried commenting out this section (that removes stray elements) and can't see any change in behaviour - can we delete it?